### ISSUE TITLE
Beta.29: Namespace pattern ie: `List.Item` breaks prop tables for sub components.

### ISSUE THREAD

**Issue by sami616** [storybookjs/storybook#11199]
When my sub component is totally separate i see a tab rendered with the correct prop table. ✅

```typescript
export default {
  title: 'UI/Elements/Accordion',
  component: Accordion,
  subcomponents: { AccordionItem },
}
```

However, if i use the popular namespace pattern: ❌

```typescript
export default {
  title: 'UI/Elements/Accordion',
  component: Accordion,
  subcomponents: { 'Accordion.Item': Accordion.Item },
}
```

I see ```No props found for this component```

**Comment by shilman** [comment]
@sami616 any idea whether this worked in earlier betas? cc @hipstersmoothie

**Comment by sami616** [comment]
@shilman sorry i have no idea!

**Comment by hipstersmoothie** [comment]
Are your components written in typescript or javascript? @sami616

**Comment by sami616** [comment]
@hipstersmoothie TypeScript!

**Comment by hipstersmoothie** [comment]
So this isn't currently supported but I just opened a PR to fix it 🎉 

https://github.com/styleguidist/react-docgen-typescript/pull/265

**Comment by alexbchr** [comment]
Using `subcomponents: { 'Accordion.Item': Accordion.Item }` worked perfectly for me with my setup. 

However, when using stories with args, the `subcomponents` property seems to be completely ignored. If I remove the args from the story, subcomponents prop tables are displayed correctly.

**Comment by shilman** [comment]
@alexbchr will fix in https://github.com/storybookjs/storybook/issues/10958

**Comment by alexbchr** [comment]
Oh nice searched for this issue and found this one instead! Thanks for the info!

**Comment by shilman** [comment]
Closing. I think this was fixed in #10958. If not, please let me know!
